### PR TITLE
Handle failed transactions when verifying donations

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
Giveth/giveth-dapps-v2#795

### CODE REVIEW

File: src/services/transactionService.ts [hunk 1]
```diff
@@ -176,9 +176,15 @@ async function getTransactionDetailForNormalTransfer(
   const transaction = await getNetworkWeb3(networkId).eth.getTransaction(
     txHash,
   );
+  const receipt = await getNetworkWeb3(networkId).eth.getTransactionReceipt(
+    txHash,
+  );
```
Comment (mohammadranjbarz) [inline src/services/transactionService.ts:181]: @CarlosQ96 
```
getNetworkWeb3(networkId).eth.getTransaction(
    txHash,
  );
  ```
  return the tx info and it doesn't matter tx is failed or not, so we have to use
  ```
   getNetworkWeb3(networkId).eth.getTransactionReceipt(
    txHash,
  );
  ```
  for this purpose

File: src/services/transactionService.ts [hunk 2]
```diff
@@ -216,6 +224,9 @@ async function getTransactionDetailForTokenTransfer(
   if (!transaction) {
     return null;
   }
+  if (!receipt?.status) {
```
Comment (CarlosQ96) [inline src/services/transactionService.ts:227]: So does receipt status is missing or the whole receipt doesn't get fetched?

**Review by CarlosQ96 — APPROVED** [review]
Hey Mohammad in general all looks good to me. Only the question I left.

File: src/services/transactionService.ts (hunk 2 above)
Comment (mohammadranjbarz) [reply src/services/transactionService.ts:227]: The `status` for failed transactions is `false` otherwise it's `true`

**Pull request merged**